### PR TITLE
bump: Update p-map package

### PR DESCRIPTION
**Pull request by JhontSouth**
#minor

## Description
This PR updates the _**p-map**_ version to the latest version and applies the necessary configuration to keep the compatibility in every library.

## Specific Changes
  - Updated _**p-map**_  to version ^7.0.3.
  - Added prebuild scripts to _botbuilder-azure_ and _botbuilder-azure-blobs_ to use tsup to compile ESM-only package.
  - Updated imports of  _**p-map**_  in every class.
  - Removed **_p-map_** from _botbuilder-repo-utils_.

## Testing
The following images show the  _botbuilder-azure_ and _botbuilder-azure-blobs_ unit tests working after the updates.
image
image

### CODE REVIEW

File: .depcheckrc [hunk 1]
```diff
@@ -1,2 +1,2 @@
-ignores: ["filenamify", "mocha", "rimraf", "sinon", "uuid"]
+ignores: ["filenamify", "mocha", "rimraf", "sinon", "uuid", "p-map"]
```
Comment (ceciliaavila) [inline .depcheckrc:1]: These were alphabetically ordered.
```suggestion
ignores: ["filenamify", "mocha", "p-map", "rimraf", "sinon", "uuid"]
```

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: .depcheckrc [hunk 2]
```diff
@@ -1,2 +1,2 @@
-ignores: ["filenamify", "mocha", "rimraf", "sinon", "uuid"]
+ignores: ["filenamify", "mocha", "p-map", "rimraf", "sinon", "uuid" ]
```
[the remark this answers is not in the document]
Comment (JhontSouth) [reply .depcheckrc]: done

**Review by ceciliaavila — APPROVED** [review]

File: libraries/botbuilder-azure-blobs/package.json [hunk 3]
```diff
@@ -30,16 +30,17 @@
...
     "@azure/core-http": "^3.0.4"
   },
   "scripts": {
     "build": "tsc -b",
     "build-docs": "typedoc --theme markdown --entryPoint botbuilder-azure-blobs --excludePrivate --includeDeclarations --ignoreCompilerErrors --module amd --out ..\\..\\doc\\botbuilder-azure-blobs .\\lib\\index.d.ts --hideGenerator --name \"Bot Builder SDK - Azure Blobs\" --readme none",
-    "clean": "rimraf _ts3.4 lib tsconfig.tsbuildinfo",
+    "clean": "rimraf _ts3.4 lib vendors tsconfig.tsbuildinfo",
     "depcheck": "depcheck --config ../../.depcheckrc",
     "lint": "eslint . --config ../../eslint.config.cjs",
+    "prebuild": "tsup ./node_modules/p-map/*.js --format cjs --dts --out-dir vendors/p-map --clean --sourcemap",
```
Comment (sw-joelmut) [inline libraries/botbuilder-azure-blobs/package.json:43]: Add the vendors folder to the files property in this package.json like...
imagen

File: libraries/botbuilder-azure/package.json [hunk 4]
```diff
@@ -47,9 +47,10 @@
     "build": "tsc -b",
     "build-docs": "typedoc --theme markdown --entryPoint botbuilder-azure --excludePrivate --includeDeclarations --ignoreCompilerErrors --module amd --out ..\\..\\doc\\botbuilder-azure .\\lib\\index.d.ts --hideGenerator --name \"Bot Builder SDK - Azure\" --readme none",
     "build:rollup": "yarn clean && yarn build && api-extractor run --verbose --local",
-    "clean": "rimraf _ts3.4 lib tsconfig.tsbuildinfo",
+    "clean": "rimraf _ts3.4 lib vendors tsconfig.tsbuildinfo",
     "depcheck": "depcheck --config ../../.depcheckrc",
     "lint": "eslint .",
+    "prebuild": "tsup ./node_modules/p-map/*.js --format cjs --dts --out-dir vendors/p-map --clean --sourcemap",
```
Comment (sw-joelmut) [inline libraries/botbuilder-azure/package.json:53]: Add the vendors folder to the files property in this package.json like...
imagen

**Review by sw-joelmut — CHANGES_REQUESTED** [review]

File: libraries/botbuilder-azure-blobs/package.json (hunk 3 above)
Comment (JhontSouth) [reply libraries/botbuilder-azure-blobs/package.json:43]: fixed

File: libraries/botbuilder-azure/package.json (hunk 4 above)
Comment (JhontSouth) [reply libraries/botbuilder-azure/package.json:53]: fixed

**Review by sw-joelmut — APPROVED** [review]

**Comment by JhontSouth** [comment]
Promoted /4820

**Pull request closed without merging**